probe/ebpf/bpf.h

```diff
@@ -28,6 +28,28 @@
 # define memmove(dest, src, n)  __builtin_memmove((dest), (src), (n))
 #endif
 
+/*
+The memcmp() built-in had some corner cases where inlining did not take place due to an LLVM issue in the back end, and is therefore not recommended to be used until the issue is fixed.
+*/
+//#define LLVM_MEMCMP_WORKING 1
+#ifdef LLVM_MEMCMP_WORKING
+#ifndef memcmp
+# define memcmp(src1, src2, n)   __builtin_memcmp((src1), (src2), (n))
+#endif
+#else
+#define memcmp discret_memcmp
+static __inline int discret_memcmp(char *s1, char *s2, int n) {
+	int i;
+#pragma unroll
+	for(i=0;i<n;s1++,s2++,i++) {
+		if (*s1 != *s2) {
```

Comment:
should we not compare s1 and s2 in order to return -1/1 in order to be sure that we provide the same behavior. Saying that because if later we expect to have the correct behavior we will have an issue.

Reply:
yeah useless in our case, but I fixed it